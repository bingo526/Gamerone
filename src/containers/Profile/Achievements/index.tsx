import React from 'react';
import { connect } from 'react-redux';
import { selectCurrentProfileUser } from 'redux/profile/selectors';
import { RootState } from 'redux/types';
import { Achievement } from 'interfaces';
import usePromise from 'lib/usePromise';
import { getProfileAchievementsById } from 'api/achievements';
import { UserModel } from 'models/user';
import Loader from 'components/common/Loader';
import AchievementFullCard from './AchievementFull';
import AchievementCard from './AchievementCard';
import { AchievementModel } from 'models/AchievementModel';
import AchievementBannerCard from './AchievementBannerCard';
import './style.scss';
import OpenGraph, { PageMeta } from 'components/common/OpenGraph';
import { SITE_URL } from 'utils/constants';
import sort from 'fast-sort';

const Achievements: React.FC<MappedProps> = ({
  user,
}: MappedProps): JSX.Element => {
  const [isLoading, achievements, error] = usePromise<AchievementModel[]>(
    () =>
      new Promise(async (resolve, reject) => {
        try {
          const data = (await getProfileAchievementsById(
            user.id,
          )) as Achievement[];
          const achivements = data.map((item) =>
            new AchievementModel().fromDto(item),
          );

          sort(achivements).desc((ach) => ach.date.getTime());

          resolve(achivements);
        } catch (ex) {
          reject(ex);
        }
      }),
    [user],
  );

  const pageTitle = user?.username + "'s Achievements";
  const achievementMeta = {
    title:
      'Check out ' +
      user?.username +
      "'s " +
      achievements?.length +
      ' achievements.',
    description: user?.bio,
    image: user?.avatar,
    url: SITE_URL + '/' + user?.username + '/achievements',
    type: 'profile',
    username: user?.username,
    firstName: user?.firstName,
    lastName: user?.lastName,
  } as PageMeta;

  return (
    <>
      <OpenGraph title={pageTitle} meta={achievementMeta} />
      <AchievementBannerCard action="follower" achievements={achievements} />
      <AchievementFullCard>
        {isLoading && !error && <Loader />}
        {!isLoading && achievements?.length === 0 && <h3>No Achievements</h3>}
        {achievements != null &&
          achievements.map((achievement) => (
            <AchievementCard key={achievement.id} achievement={achievement} />
          ))}
      </AchievementFullCard>
      <AchievementBannerCard action="post" achievements={achievements} />
    </>
  );
};

const mapStateToProps = (state: RootState) => {
  return {
    user: selectCurrentProfileUser(state) || new UserModel(),
  };
};

type MappedProps = ReturnType<typeof mapStateToProps>;

export default connect(mapStateToProps)(Achievements);
