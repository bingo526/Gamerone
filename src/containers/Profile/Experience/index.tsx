import React, { useState, useEffect } from 'react';
import Card, { CardTypeEnum } from 'components/common/Card';
import { UserExperience, ExperienceType } from 'interfaces';
import { useSelector, useDispatch } from 'react-redux';
import { Notify } from 'components/utility/Notify';
import * as ReferenceDataApi from 'api/referenceData';
import ExperienceSettings from './ExperienceSettings';
import './style.scss';
import Loader from 'components/common/Loader';
import ExperienceItem from './ExperienceItem';
import {
  selectIsSelfProfile,
  selectExperience,
  selectExperienceStatus,
} from 'redux/selectors';
import DialogActions from 'redux/dialogs/actions';
import { DialogTypeEnum } from 'redux/dialogs/types';
import { SelectOption } from 'components/common/Form/Input';
import Button from 'components/common/Button';
import { LOAD_EXPERIENCES } from 'redux/types';
import ExperienceSummary from './ExperienceSummary';

export function useExperiences() {
  const experiences = useSelector(selectExperience);
  const experiencesStatus = useSelector(selectExperienceStatus);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch({
      type: LOAD_EXPERIENCES,
    });
  }, [dispatch]);

  return [experiences, experiencesStatus, dispatch] as const;
}

const Experiences: React.FC = (): JSX.Element => {
  const [experiences, experiencesStatus, dispatch] = useExperiences();
  const isOwner = useSelector(selectIsSelfProfile);

  // Experience Type
  // Loading here as workaround for FormInput select on edit
  const [ExperienceTypeOptions, setExperienceTypeOptions] = useState<
    SelectOption[]
  >([]);

  useEffect(() => {
    const fetchExperienceTypes = async () => {
      try {
        const experienceTypes = ((await ReferenceDataApi.experienceTypes()) as unknown) as ExperienceType[];

        setExperienceTypeOptions(
          experienceTypes.map((item) => ({
            label: item.name,
            value: item.id,
          })),
        );
      } catch (err) {
        Notify.warning('Unable to load experience types.');
      }
    };
    fetchExperienceTypes();
  }, []);

  // Form Dialog Handlers
  const handleAddClick = () => {
    dispatch(
      DialogActions.showDialog(DialogTypeEnum.SETTINGS_EXPERIENCE, true),
    );
  };

  const handleSetCurrentExperience = (experience: UserExperience) => {
    dispatch(
      DialogActions.showDialog(
        DialogTypeEnum.SETTINGS_EXPERIENCE,
        true,
        experience,
      ),
    );
  };

  return (
    <>
      <ExperienceSummary />
      <Card
        type={CardTypeEnum.EXPERIENCE}
        isOwner={isOwner}
        editIcon="icon-add"
        onEdit={handleAddClick}
      >
        <h4 className="card__header card__title">Gaming Experience</h4>
        <div className="experience-items">
          {experiencesStatus?.isFetching && !experiences && (
            <div
              className="experience-item"
              style={{
                display: 'flex',
                flex: 1,
                flexDirection: 'column',
                alignItems: 'center',
              }}
            >
              <Loader />
            </div>
          )}
          {experiencesStatus?.isSuccess && experiences?.length === 0 && (
            <div
              className="experience-item empty"
              style={{
                display: 'flex',
                flex: 1,
                flexDirection: 'column',
                alignItems: 'center',
              }}
            >
              <p>No Experience yet</p>

              {isOwner && (
                <Button
                  onClick={handleAddClick}
                  data-testid="add-product"
                  iconLeft="icon-single-neutral-id-card-1"
                >
                  Add experience
                </Button>
              )}
            </div>
          )}
          {experiences?.map((exp: UserExperience) => {
            return (
              <ExperienceItem
                key={exp.id}
                experience={exp}
                isOwner={isOwner}
                onEdit={handleSetCurrentExperience}
              />
            );
          })}
        </div>
      </Card>
      {isOwner && (
        <>
          <ExperienceSettings experienceTypeOptions={ExperienceTypeOptions} />
        </>
      )}
    </>
  );
};

export default Experiences;
