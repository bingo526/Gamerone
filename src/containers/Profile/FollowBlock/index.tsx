/* eslint-disable react-hooks/exhaustive-deps */
import React, { memo, useContext, useMemo, useCallback } from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';

import Button, { ButtonSchemeEnum } from 'components/common/Button';
import { selectCurrentProfileUser } from 'redux/profile/selectors';
import ProfileActions from 'redux/profile/actions';
import SettingsActions from 'redux/settings/actions';
import { AuthContext } from 'provider/auth';
import { RootState } from 'redux/types';
import { UserModel } from 'models/user';
import Dropdown, { DropdownOptionType } from 'components/common/Dropdown';
import { UserSquadStatusEnum } from 'interfaces';

export interface FollowBlockProps {
  isSelf: boolean;
  className?: string;
}

function FollowBlock({
  isSelf,
  user,
  dispatchFollow,
  dispatchUnfollow,
  dispatchBlock,
  dispatchAddSquad,
  dispatchRemoveSquad,
}: FollowBlockProps & ConnectedProps) {
  const { isAuthenticated } = useContext(AuthContext);
  let { isFollowing } = (user as UserModel).followStatus;
  const { isBlocking = false } = (user as UserModel).followStatus;

  const handleFollowClick = useCallback(
    (e: React.MouseEvent<HTMLButtonElement>) => {
      e.preventDefault();
      if (!isFollowing) {
        dispatchFollow();
        isFollowing = true;
      }
    },
    [isFollowing],
  );

  const handleUnfollowClick = useCallback(() => {
    if (isFollowing) dispatchUnfollow();
  }, [isFollowing]);

  const dropdownOptions = useMemo(() => {
    const options: DropdownOptionType[] = [];

    if (isFollowing) {
      options.push({
        label: 'Unfollow',
        onClick: (): void => handleUnfollowClick(),
      });
    }
    if (user?.isSquad === UserSquadStatusEnum.NO_SQUAD) {
      options.push({
        label: 'Invite to Squad',
        onClick: () => dispatchAddSquad(),
      });
    } else if (user?.isSquad === UserSquadStatusEnum.IN_SQUAD) {
      options.push({
        label: 'Remove from Squad',
        onClick: () => dispatchRemoveSquad(user.id),
      });
    } else {
      options.push({ label: 'Waiting for accept', disabled: true });
    }

    options.push({ label: 'Block', onClick: () => dispatchBlock() });
    options.push({ label: 'Unblock' });

    return options;
  }, [handleUnfollowClick, isFollowing]);

  return !isSelf ? (
    <>
      <Button
        scheme={!isFollowing ? ButtonSchemeEnum.PRIMARY : null}
        onClick={handleFollowClick}
        aria-label="follow-button"
        disabled={!isAuthenticated || isFollowing || isBlocking}
      >
        {isFollowing ? 'Following' : 'Follow'}
      </Button>
      <div className="last" style={{ marginLeft: 'auto' }}>
        {isAuthenticated && (
          <Dropdown options={dropdownOptions}>
            <Button
              scheme={ButtonSchemeEnum.SQUARE}
              disabled={!isAuthenticated}
            >
              ...
            </Button>
          </Dropdown>
        )}
      </div>
    </>
  ) : (
    <></>
  );
}

const mapStateToProps = (state: RootState) => ({
  user: selectCurrentProfileUser(state),
});

const mapDispatchToProps = {
  dispatchFollow: ProfileActions.follow,
  dispatchUnfollow: ProfileActions.unfollow,
  dispatchBlock: ProfileActions.block,
  dispatchAddSquad: ProfileActions.addSquad,
  dispatchRemoveSquad: SettingsActions.deleteSquad,
};

type ConnectedProps = ReturnType<typeof mapStateToProps> &
  typeof mapDispatchToProps;

const withConnect = connect(mapStateToProps, mapDispatchToProps);

export default compose(withConnect, memo)(FollowBlock) as React.ElementType;
