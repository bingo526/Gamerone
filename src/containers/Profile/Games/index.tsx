import React, { useEffect, useState, useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import InfiniteScroll from 'react-infinite-scroll-component';

import {
  selectIsSelfProfile,
  selectGamesStatus,
  selectGames,
  selectHasMoreGames,
  selectProfileUser,
} from 'redux/selectors';
import Loader from 'components/common/Loader';
import UserGameCard from './UserGameCard';
import Button, { ButtonSchemeEnum } from 'components/common/Button';
import GameSettings from './GameSettings';
import Card from 'components/common/Card';
import { LOAD_INITIAL_GAMES, LOAD_NEXT_GAMES } from 'redux/types';
import OpenGraph, { PageMeta } from 'components/common/OpenGraph';
import { SITE_URL } from 'utils/constants';
import { UserGame } from 'interfaces';
import DialogActions from 'redux/dialogs/actions';
import { DialogTypeEnum } from 'redux/dialogs/types';
import { selectGamePlatforms } from 'redux/settings/selectors';
import { SelectOption } from 'components/common/Form/FormInput';
import SettingsActions from 'redux/settings/actions';

export function useGames() {
  const games = useSelector(selectGames);
  const status = useSelector(selectGamesStatus);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch({
      type: LOAD_INITIAL_GAMES,
    });
  }, [dispatch]);

  return [games, status, dispatch] as const;
}

const Games: React.FC = (): JSX.Element => {
  const [games, status, dispatch] = useGames();
  const isOwner = useSelector(selectIsSelfProfile);
  const hasMoreGames = useSelector(selectHasMoreGames);

  const fetchMoreData = useCallback(() => {
    dispatch({
      type: LOAD_NEXT_GAMES,
    });
  }, [dispatch]);

  const gamePlatforms = useSelector(selectGamePlatforms);
  const [gamePlatformOptions, setGamePlatformOptions] = useState<
    SelectOption[]
  >([]);

  // Get Game Platforms
  useEffect(() => {
    dispatch(SettingsActions.getGamePlatforms());
    // eslint-disable-next-line
  }, []);

  // Build Select options
  useEffect(() => {
    if (gamePlatforms) {
      const mappedPlatforms: SelectOption[] = gamePlatforms.map((platform) => {
        return { value: platform.id.toString(), label: platform.name };
      });
      setGamePlatformOptions(mappedPlatforms);
    }
  }, [gamePlatforms]);

  const handleAddClick = () => {
    dispatch(DialogActions.showDialog(DialogTypeEnum.SETTINGS_GAME, true));
  };

  const handleSetCurrentUserGame = (userGame: UserGame) => {
    dispatch(
      DialogActions.showDialog(DialogTypeEnum.SETTINGS_GAME, true, userGame),
    );
  };

  // Page Meta
  const user = useSelector(selectProfileUser);
  const pageTitle = user?.username + "'s Games";
  const gamesMeta = {
    title: 'Check out ' + user?.username + "'s " + games?.length + ' games',
    description:
      user?.username +
      ' has a total of ' +
      games?.length +
      ' games listed on their Gamer One profile',
    image: user?.avatar,
    url: SITE_URL + '/' + user?.username + '/games',
    type: 'profile',
    username: user?.username,
    firstName: user?.firstName,
    lastName: user?.lastName,
  } as PageMeta;

  return (
    <>
      <OpenGraph title={pageTitle} meta={gamesMeta} />
      <div className="games-meta">
        <Card type="flat">
          <h4 className="card__header card__title">Games</h4>
          <div className="card__content"></div>
          {isOwner && (
            <div className="card__actions">
              <Button
                scheme={ButtonSchemeEnum.PRIMARY}
                iconLeft="icon-video-game-pacman"
                onClick={handleAddClick}
                data-testid="add"
              >
                Add Game
              </Button>
            </div>
          )}
        </Card>
      </div>

      {status?.isFetching && games.length === 0 && (
        <div style={{ gridColumn: '2 / span 4', marginTop: '6rem', zIndex: 1 }}>
          <h1 style={{ textAlign: 'center', fontSize: '1.5rem' }}>
            <Loader />
          </h1>
        </div>
      )}

      {status?.isSuccess && games.length === 0 && (
        <div style={{ gridColumn: '2 / span 4', marginTop: '6rem', zIndex: 1 }}>
          <h1 style={{ textAlign: 'center', fontSize: '1.5rem' }}>No Games.</h1>
        </div>
      )}
      <InfiniteScroll
        dataLength={games.length}
        next={fetchMoreData}
        hasMore={hasMoreGames}
        loader={Loader}
        className="infinite-grid"
        style={{ overflow: 'inherit' }}
      >
        {games.map((userGame) => {
          return (
            <UserGameCard
              key={userGame.id}
              usergame={userGame}
              isOwner={isOwner}
              onEditClick={handleSetCurrentUserGame}
            />
          );
        })}
      </InfiniteScroll>

      {isOwner && <GameSettings platformOptions={gamePlatformOptions} />}
    </>
  );
};

export default Games;
