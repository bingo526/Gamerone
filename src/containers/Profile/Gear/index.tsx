import React, { useEffect, useCallback, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import ProfileActions from 'redux/profile/actions';
import SettingsActions from 'redux/settings/actions';

import {
  selectGears,
  selectIsSelfProfile,
  selectGearsStatus,
  selectProfileUser,
} from 'redux/selectors';
import { selectCurrentProfileId } from 'redux/profile/selectors';
import Loader from 'components/common/Loader';
import UserGearCard from './UserGearCard';
import Button, { ButtonSchemeEnum } from 'components/common/Button';
import GearSettings from '../Overview/Dialogs/GearSettings';
import Card from 'components/common/Card';
import OpenGraph, { PageMeta } from 'components/common/OpenGraph';
import { SITE_URL } from 'utils/constants';

export function useGears() {
  const isOwner = useSelector(selectIsSelfProfile);
  const userId = useSelector(selectCurrentProfileId);
  const gears = useSelector(selectGears);
  const status = useSelector(selectGearsStatus);
  const dispatch = useDispatch();

  useEffect(() => {
    if (userId) {
      dispatch(
        isOwner
          ? SettingsActions.loadInitialPage('gears')
          : ProfileActions.loadInitialPage('gears'),
      );
    }

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [userId]);

  return [gears, status] as const;
}

const Gear: React.FC = (): JSX.Element => {
  const [gears, status] = useGears();
  const isOwner = useSelector(selectIsSelfProfile);
  const [gearSettingsIsOpen, setGearSettingsIsOpen] = useState(false);

  const handleClickGearEdit = useCallback(() => {
    setGearSettingsIsOpen(true);
  }, []);

  const handleCloseGearSettings = useCallback(() => {
    setGearSettingsIsOpen(false);
  }, []);

  // Page Meta
  const user = useSelector(selectProfileUser);
  const pageTitle = user?.username + "'s Gear";
  const gearMeta = {
    title: 'Check out ' + user?.username + "'s " + gears?.length + ' items',
    description:
      user?.username +
      ' has a total of ' +
      gears?.length +
      ' hardware items listed on their Gamer One profile',
    image: user?.avatar,
    url: SITE_URL + '/' + user?.username + '/gear',
    type: 'profile',
    username: user?.username,
    firstName: user?.firstName,
    lastName: user?.lastName,
  } as PageMeta;

  return (
    <>
      <OpenGraph title={pageTitle} meta={gearMeta} />
      <div className="games-meta">
        <Card type="flat">
          <h4 className="card__header card__title">Gear</h4>
          <div className="card__content"></div>
          {isOwner && (
            <div className="card__actions">
              <Button
                scheme={ButtonSchemeEnum.PRIMARY}
                iconLeft="icon-keyboard"
                onClick={handleClickGearEdit}
                data-testid="add"
              >
                Manage
              </Button>
            </div>
          )}
        </Card>
      </div>

      {status?.isFetching && gears.length === 0 && (
        <div style={{ gridColumn: '2 / span 4', marginTop: '6rem', zIndex: 1 }}>
          <h1 style={{ textAlign: 'center', fontSize: '1.5rem' }}>
            <Loader />
          </h1>
        </div>
      )}

      {status?.isSuccess && gears.length === 0 && (
        <div style={{ gridColumn: '2 / span 4', marginTop: '6rem', zIndex: 1 }}>
          <h1 style={{ textAlign: 'center', fontSize: '1.5rem' }}>No Gear.</h1>
        </div>
      )}

      {gears.map((userGear) => {
        return <UserGearCard key={userGear.id} userGear={userGear} />;
      })}
      {isOwner && (
        <GearSettings
          visible={gearSettingsIsOpen}
          onClose={handleCloseGearSettings}
        />
      )}
    </>
  );
};

export default Gear;
