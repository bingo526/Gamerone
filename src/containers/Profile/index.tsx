import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  Switch,
  Route as ReactRoute,
  useHistory,
  useParams,
  useRouteMatch,
} from 'react-router-dom';

import Grid from 'components/layout/Grid';
import withChrome from 'components/common/Chrome/withChrome';
import NotFound from 'containers/NotFound';
import { OverviewRoute, ProfileRoute } from './route';
import ProfileNav from './ProfileNav';

import { selectIsSelfProfile, selectProfileUser } from 'redux/selectors';

import ProfileActions from 'redux/profile/actions';
import { Route } from 'interfaces';
import * as ProfileApi from 'api/profile';
import usePromise from 'lib/usePromise';
import ProfileCover from './ProfileCover';
import { ProfileLayoutProcessTypeEnum } from 'redux/profile/types';

import UserCard from 'containers/Cards/UserCard';
import PostDetail from 'containers/Post/PostDetail';
import Page from 'components/layout/Page';
import LoaderFullScreen from 'components/common/LoaderFullScreen';

const ChromeGrid = withChrome(Grid);

const Profile: React.FC = (): JSX.Element => {
  const [isEditing, setIsEditing] = useState(false);
  const user = useSelector(selectProfileUser);
  const isSelf = useSelector(selectIsSelfProfile);
  const history = useHistory();
  const dispatch = useDispatch();

  const { username } = useParams();
  const [nameResolving, resolvedProfile, nameError] = usePromise<Route>(
    () => ProfileApi.resolveRoute(username),
    [username],
  );

  const onEditLayout = () => {
    dispatch(
      ProfileActions.getProfileLayoutProcess(
        ProfileLayoutProcessTypeEnum.IsEdit,
      ),
    );
    setIsEditing(true);
  };

  const onSaveLayout = () => {
    dispatch(
      ProfileActions.getProfileLayoutProcess(ProfileLayoutProcessTypeEnum.Save),
    );
    setIsEditing(false);
  };

  const onMakeLayoutDefault = () => {
    dispatch(
      ProfileActions.getProfileLayoutProcess(
        ProfileLayoutProcessTypeEnum.Default,
      ),
    );
    setIsEditing(false);
  };

  const onCancelEdit = () => {
    dispatch(
      ProfileActions.getProfileLayoutProcess(
        ProfileLayoutProcessTypeEnum.Cancel,
      ),
    );
    setIsEditing(false);
  };

  useEffect(() => {
    if (resolvedProfile) {
      dispatch(ProfileActions.setResolvedContent(resolvedProfile));
    }
    if (nameError) dispatch(ProfileActions.setCurrentProfile(null));
  }, [resolvedProfile, nameError, dispatch]);

  return (
    <>
      {nameResolving && !resolvedProfile && <LoaderFullScreen />}
      {!nameResolving && nameError && <NotFound username={username} />}
      {!nameResolving && resolvedProfile && (
        <>
          <ProfileCover />
          <ChromeGrid>
            {user &&
              (history.location.pathname === `/${user.username}` ? (
                <UserCard
                  user={user}
                  isSelf={isSelf}
                  allowEditLayout={true}
                  isEditing={isEditing}
                  onEdit={onEditLayout}
                  onSaveLayoutEdit={onSaveLayout}
                  onDefault={onMakeLayoutDefault}
                  onCancelEdit={onCancelEdit}
                />
              ) : (
                <UserCard type="short" user={user} isSelf={isSelf} />
              ))}
            <ProfileNav />
            <ProfileRoute />
          </ChromeGrid>
          <OverviewRoute />
        </>
      )}
    </>
  );
};

const ProfilePost = (): JSX.Element => {
  const match = useRouteMatch();
  // const { username } = useParams();
  return (
    <Page>
      <Switch>
        <ReactRoute
          exact
          path={`${match.path}/post/:postId(\\d+)`}
          component={withChrome(PostDetail, 200)}
        />
        <ReactRoute path={`${match.path}`} component={Profile} />
      </Switch>
    </Page>
  );
};

export default ProfilePost;
