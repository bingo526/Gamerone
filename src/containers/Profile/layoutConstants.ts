/**
 * breakpoints and number of columns for each screen size
 * desktop: 5
 * tablet: 3
 * mobile: 1
 */

export const PROFILE_LAYOUT_BREAKPOINTS = {
  lg: 1200,
  md: 996,
  sm: 768,
  xs: 480,
};
export const PROFILE_LAYOUT_COLUMNS = { lg: 5, md: 3, sm: 3, xs: 1 };

// initial layouts for react-grid-layout (desktop, tablet, mobile)
export const PROFILE_LAYOUT_DEFAULT = {
  lg: [
    { i: 'block-user', x: 0, y: 0, w: 2, h: 2, static: true },
    { i: 'block-nav', x: 2, y: 0, w: 3, h: 1, static: true },
    {
      i: 'block-sponsors',
      x: 2,
      y: 1,
      w: 3,
      h: 1,
      minW: 1,
      maxW: 4,
      minH: 1,
      maxH: 1,
    },
    {
      i: 'block-latest-post',
      x: 0,
      y: 2,
      w: 3,
      h: 1.5,
      minW: 2,
      maxW: 3,
      minH: 1.5,
      maxH: 1.5,
    },
    {
      i: 'block-now-playing',
      x: 3,
      y: 2,
      w: 2,
      h: 1.5,
      minW: 2,
      maxW: 2,
      minH: 1.5,
      maxH: 1.5,
    },
    {
      i: 'block-social-networks',
      x: 0,
      y: 4,
      w: 1,
      h: 1,
      minW: 1,
      maxW: 2,
      minH: 1,
      maxH: 1,
    },
    {
      i: 'block-achievements',
      x: 1,
      y: 4,
      w: 1,
      h: 1,
      minW: 1,
      maxW: 2,
      minH: 1,
      maxH: 1,
    },
    {
      i: 'block-gears',
      x: 2,
      y: 4,
      w: 2,
      h: 1,
      minW: 1,
      maxW: 3,
      minH: 1,
      maxH: 1,
    },

    {
      i: 'block-store',
      x: 4,
      y: 4,
      w: 1,
      h: 2,
      minW: 1,
      maxW: 2,
      minH: 2,
      maxH: 2,
    },
    {
      i: 'block-squad',
      x: 0,
      y: 6,
      w: 2,
      h: 1,
      minW: 1,
      maxW: 2,
      minH: 1,
      maxH: 1,
    },
  ],
  md: [
    { i: 'block-user', x: 0, y: 0, w: 2, h: 2, static: true },
    { i: 'block-nav', x: 2, y: 0, w: 1, h: 3, static: true },
    { i: 'block-sponsors', x: 0, y: 2, w: 3, h: 1, minW: 2, minH: 1, maxH: 1 },
    { i: 'block-achievements', x: 0, y: 3, w: 1, h: 2, minH: 2, maxH: 2 },
    {
      i: 'block-latest-post',
      x: 1,
      y: 3,
      w: 2,
      h: 2,
      minW: 2,
      minH: 2,
      maxH: 2,
    },
    { i: 'block-gears', x: 0, y: 5, w: 3, h: 1, minW: 2, minH: 1, maxH: 1 },
    { i: 'block-social-networks', x: 0, y: 8, w: 1, h: 1, minH: 1, maxH: 1 },
    { i: 'block-now-playing', x: 1, y: 8, w: 2, h: 2, minH: 2, maxH: 2 },
    { i: 'block-store', x: 4, y: 6, w: 1, h: 2, minH: 2, maxH: 2 },
    { i: 'block-squad', x: 0, y: 6, w: 1, h: 2, minW: 1, minH: 1, maxH: 2 },
  ],
  sm: [
    { i: 'block-user', x: 0, y: 0, w: 3, h: 2, static: true },
    { i: 'block-nav', x: 0, y: 2, w: 3, h: 1, static: true },
    { i: 'block-sponsors', x: 0, y: 3, w: 3, h: 1, minW: 2, minH: 1, maxH: 1 },
    { i: 'block-achievements', x: 0, y: 4, w: 1, h: 2, minH: 2, maxH: 2 },
    {
      i: 'block-latest-post',
      x: 1,
      y: 4,
      w: 2,
      h: 2,
      minW: 2,
      minH: 2,
      maxH: 2,
    },
    { i: 'block-gears', x: 0, y: 6, w: 3, h: 1, minW: 2, minH: 1, maxH: 1 },
    { i: 'block-social-networks', x: 0, y: 9, w: 1, h: 1, minH: 1, maxH: 1 },
    { i: 'block-now-playing', x: 1, y: 9, w: 2, h: 2, minH: 2, maxH: 2 },
    { i: 'block-store', x: 4, y: 6, w: 1, h: 2, minH: 2, maxH: 2 },
    { i: 'block-squad', x: 0, y: 6, w: 1, h: 2, minW: 1, minH: 1, maxH: 2 },
  ],
  xs: [
    { i: 'block-user', x: 0, y: 0, w: 1, h: 2, static: true },
    { i: 'block-nav', x: 0, y: 2, w: 1, h: 1, static: true },
    { i: 'block-sponsors', x: 0, y: 4, w: 1, h: 1, minH: 1, maxH: 1 },
    { i: 'block-achievements', x: 0, y: 4, w: 1, h: 2, minH: 2, maxH: 2 },
    {
      i: 'block-latest-post',
      x: 0,
      y: 6,
      w: 1,
      h: 2,
      minW: 2,
      minH: 2,
      maxH: 2,
    },
    { i: 'block-gears', x: 0, y: 8, w: 1, h: 1, minH: 1, maxH: 1 },
    { i: 'block-social-networks', x: 0, y: 9, w: 1, h: 1, minH: 1, maxH: 1 },
    { i: 'block-now-playing', x: 0, y: 12, w: 1, h: 2, minH: 2, maxH: 2 },
    { i: 'block-store', x: 4, y: 6, w: 1, h: 2, minH: 2, maxH: 2 },
    { i: 'block-squad', x: 0, y: 6, w: 1, h: 2, minW: 1, minH: 1, maxH: 2 },
  ],
};
