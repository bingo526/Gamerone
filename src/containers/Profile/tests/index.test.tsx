import React from 'react';
import { Provider } from 'react-redux';
import { Router, Route } from 'react-router-dom';
import { createProduct } from 'redux';
import { render, waitForElement } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { act } from 'react-dom/test-utils';
import profileReducer from 'redux/profile/reducer';
import Profile from '../index';
import { mockFetchError } from './utils';

const renderWithProfileReducer = (
  component = <Profile />,
  route = '/',
  {
    initialState,
    store = createProduct(profileReducer, { Profile: initialState }),
  } = {},
) => {
  const history = createMemoryHistory({ initialEntries: [route] });

  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>
          <Route path="/:username" render={() => component} />
        </Router>
      </Provider>,
    ),
    history,
  };
};

describe('page render', () => {
  test('shows user is not found', async () => {
    const username = 'vadim';
    const errorMessage = `${username} is not found`;

    const spy = mockFetchError(404);

    await act(async () => {
      const { getByText, asFragment } = renderWithProfileReducer(
        <Profile />,
        '/' + username,
        {
          initialState: {
            currentProfile: null,
          },
        },
      );

      await waitForElement(() => getByText(errorMessage));
      expect(asFragment()).toMatchSnapshot();
      expect(getByText(errorMessage)).toBeInTheDocument();

      spy.mockRestore();
    });
  });
});
