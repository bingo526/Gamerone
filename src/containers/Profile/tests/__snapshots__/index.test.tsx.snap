// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page render shows user is not found 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div
      class="alert"
    >
      <h1>
        vadim is not found
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
